photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala

```diff
@@ -63,12 +64,22 @@ final class Driver(val sc: SparkContext, val params: GameParams, implicit val lo
     val featureIndexMapLoaders = Timed("prepare features") {
       prepareFeatureMaps()
     }
-    val trainingData = Timed("read training data") {
+    val trainingDataToVerify = Timed("read training data") {
       readTrainingData(featureIndexMapLoaders)
     }
-    val validationData = Timed("read validation data") {
+
+    val trainingData = Timed("verify training data") {
+      checkData(trainingDataToVerify)
```

Comment:
The name `checkData` is a little misleading here, because this is actually modifying in addition to checking.